packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,149 @@
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * A single tag property values
+ */
+interface ITagProps {
+    value: string | Token;
+    propagate?: boolean;
+}
+
+/**
+ * A an object of tags
+ */
+export interface ITags {
+    [key: string]: ITagProps;
+}
+
+/**
+ * A single tag
+ */
+export interface ITag {
+    key: string;
+    value: string | Token;
+}
+
+/**
+ * TagManager facilitates a common implementation of tagging for Constructs.
+ *
+ * Each construct that wants to support tags should implement the `ITaggable`
+ * interface and properly pass tags to the `Resources` (Cloudformation) elements
+ * the `Construct` creates using `toToken()` for lazy evaluations
+ */
+export class TagManager {
+
+    /**
+     * Checks if the object implements the `ITaggable` interface
+     */
+    public static isTaggable(taggable: ITaggable | any): taggable is ITaggable {
+        return ((taggable as ITaggable).tags !== undefined);
+    }
+
+    private readonly _tags: ITags = {};
+
+    private readonly parentTagManager: TagManager | undefined;
+
+    constructor(partner: Construct, initialTags: ITags = {}) {
```

Comment:
Doesn't look like `partner` is used at all. Why not just pass `parentConstruct` here? This will also allow you to create TagManagers for the child-resources of the VPC without needing L2s for them (see comment above).

Reply:
great insight -- this helped clean some stuff up.